src/solvers/refinement/string_refinement.h

```diff
@@ -27,57 +27,11 @@ Author: Alberto Griggio, alberto.griggio@gmail.com
 #include <solvers/refinement/string_constraint.h>
 #include <solvers/refinement/string_constraint_generator.h>
 #include <solvers/refinement/string_refinement_invariant.h>
+#include <solvers/refinement/string_refinement_util.h>
```

Comment:
Do you actually need this in the header file?

Reply:
Not easy to avoid because some fields have type defined there.

Reply:
If it's unavoidable then that's what it is, but if forward declarations would do the trick (as may be the case with the other includes) then it would help reducing dependencies during compilation. Just something to keep in mind.